week-7/Homework/mandatory/1-debugging-practice/script.js

```diff
@@ -28,76 +28,83 @@ const check = document.getElementById("check");
-    render();
-  }
+    if (
+        title.value == null ||
+        title.value == "" ||
+        pages.value == null ||
+        pages.value == "" ||
+        author.value == null ||
+        author.value == ""
+    ) {
+        alert("Please fill all fields!");
+        return false;
+    } else {
+        let book = new Book(title.value, author.value, pages.value, check.checked);
+        myLibrary.push(book);
+        render();
+        title.value = "";
+        author.value = "";
+        pages.value = "";
+        check.checked = false;
+
+    }
 }
 
 function Book(title, author, pages, check) {
-  this.title = title;
-  this.author = author;
-  this.pages = pages;
-  this.check = check;
+    this.title = title;
+    this.author = author;
+    this.pages = pages;
+    this.check = check;
 }
 
 function render() {
-  let table = document.getElementById("display");
-  let rowsNumber = table.rows.length;
-  //delete old table
-  for (let n = rowsNumber - 1; n > 0; n-- {
-    table.deleteRow(n);
-  }
-  //insert updated row and cells
-  let length = myLibrary.length;
-  for (let i = 0; i < length; i++) {
-    let row = table.insertRow(1);
-    let cell1 = row.insertCell(0);
-    let cell2 = row.insertCell(1);
-    let cell3 = row.insertCell(2);
-    let cell4 = row.insertCell(3);
-    let cell5 = row.insertCell(4);
-    cell1.innerHTML = myLibrary[i].title;
-    cell2.innerHTML = myLibrary[i].author;
-    cell3.innerHTML = myLibrary[i].pages;
-
-    //add and wait for action for read/unread button
-    let changeBut = document.createElement("button");
-    changeBut.id = i;
-    changeBut.className = "btn btn-success";
-    cell4.appendChild(changeBut);
-    let readStatus = "";
-    if (myLibrary[i].check == false) {
-      readStatus = "Yes";
-    } else {
-      readStatus = "No";
+    let table = document.getElementById("display");
+    let rowsNumber = table.rows.length;
+    //delete old table
+    for (let n = rowsNumber - 1; n > 0; n--) {
+        table.deleteRow(n);
     }
-    changeBut.innerHTML = readStatus;
+    //insert updated row and cells
+    let length = myLibrary.length;
+    for (let i = 0; i < length; i++) {
+        let row = table.insertRow(1);
+        let cell1 = row.insertCell(0);
+        let cell2 = row.insertCell(1);
+        let cell3 = row.insertCell(2);
+        let cell4 = row.insertCell(3);
+        let cell5 = row.insertCell(4);
+        cell1.innerHTML = myLibrary[i].title;
+        cell2.innerHTML = myLibrary[i].author;
+        cell3.innerHTML = myLibrary[i].pages;
 
-    changeBut.addEventListener("click", function () {
-      myLibrary[i].check = !myLibrary[i].check;
-      render();
-    });
+        //add and wait for action for read/unread button
+        let changeBut = document.createElement("button");
+        changeBut.id = i;
+        changeBut.className = "btn btn-success";
+        cell4.appendChild(changeBut);
+        let readStatus = "";
+        if (myLibrary[i].check == false) {
```

Comment:
this is for sure correct
for more refactored code you can use something like 
`if(myLibrary[i].check){readStatus="yes"}else{readStatus="no"}`
or set it to "no" outside and set it to yes only in one condition 
`let readStatus = 'No'
if(myLibrary[i].check){readStatus="yes"}`